nirtorch/graph.py

```diff
@@ -34,20 +37,20 @@ def __init__(
         self,
         elem: Any,
         name: str,
-        outgoing_nodes: Optional[List["Node"]] = None,
+        outgoing_nodes: Optional[Dict["Node", torch.Tensor]] = None,
```

Comment:
Why are outgoing Nodes allowed to be Tensors? Ideally if they need to connect to tensors, they should also be wrapped into a Node.

Reply:
Outgoing nodes should not be tensors, but it would be helpful if they have a shape. Hence, the `torch.Tensor` value in the dictionary. Come to think of it, it may be better if we encode it as a `torch.Size`.

Depending on the [NIR PR on subgraphs](https://github.com/neuromorphs/NIR/pull/39), we should maybe even reconsider the type to be something like a tuple of shapes (e. g. `torch.Size`).